packages/account/test/block-sensitive-operations.test.ts

```diff
@@ -0,0 +1,240 @@
+import { hexlify, sleep } from '@fuel-ts/utils';
+import type { MockInstance } from 'vitest';
+
+import { Provider, type WalletUnlocked } from '../src';
+import { setupTestProviderAndWallets } from '../src/test-utils';
+
+async function createTransfer(wallet: WalletUnlocked) {
+  const baseAssetId = await wallet.provider.getBaseAssetId();
+
+  const tx = await wallet.createTransfer(wallet.address, '10', baseAssetId);
+  const signature = await wallet.signTransaction(tx);
+  tx.updateWitnessByOwner(wallet.address, signature);
+
+  const encodedTransaction = hexlify(tx.toTransactionBytes());
+  const transactionId = tx.getTransactionId(await wallet.provider.getChainId());
+  return {
+    encodedTransaction,
+    transactionId,
+  };
+}
+
+async function setupTest() {
+  const {
+    provider,
+    wallets: [wallet],
+    cleanup,
+  } = await setupTestProviderAndWallets({
+    nodeOptions: {
+      /**
+       * selected 500ms interval and not something lower
+       * because of the flakiness experienced
+       * where the getLatestBlockHeight query would return
+       * a block height that is lower than the one returned in the extensions object.
+       */
+      args: ['--poa-instant', 'false', '--poa-interval-period', '500ms'],
+    },
+  });
+
+  // allow for block production
+  await sleep(550);
+
+  const { encodedTransaction, transactionId } = await createTransfer(wallet);
+
+  const {
+    chain: {
+      latestBlock: { height: latestBlockHeight },
+    },
+  } = await provider.operations.getLatestBlockHeight();
+
+  return {
+    provider,
+    wallet,
+    latestBlockHeight: +latestBlockHeight,
+    encodedTransaction,
+    transactionId,
+    [Symbol.dispose]: cleanup,
+  };
+}
+
+type ExpectBlockHeightProps = { fetchSpy: MockInstance<typeof fetch> } & (
+  | { included: true; latestBlockHeight: number }
+  | { included: false }
+);
+
+function expectBlockHeight(props: ExpectBlockHeightProps) {
+  const body = JSON.parse(props.fetchSpy.mock.calls[0][1]?.body?.toString() as string);
+  const property = 'extensions.required_fuel_block_height';
+
+  if (props.included) {
+    expect(body).toHaveProperty(property, props.latestBlockHeight);
+  } else {
+    expect(body).not.toHaveProperty(property);
+  }
+}
+
+/**
+ * @group node
+ * @group browser
+ */
+describe('Block-sensitive operations have the current block height included in request', () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+    Provider.ENSURE_RPC_CONSISTENCY = true;
+  });
+
+  test('submit - block height included', async () => {
+    using launched = await setupTest();
+
+    const { provider, encodedTransaction, latestBlockHeight } = launched;
+
+    const fetchSpy = vi.spyOn(global, 'fetch');
+
+    await provider.operations.submit({ encodedTransaction });
+
+    expectBlockHeight({ included: true, fetchSpy, latestBlockHeight });
+  });
```

Comment:
Do you think we should add test cases here with Providers that were instantiated with custom fetch fn?

Reply:
The existing tests still cover the custom fetch fn functionality and would fail if [this line](https://github.com/FuelLabs/fuels-ts/blob/ns/feat/optimistic-concurrency-control/packages/account/src/providers/provider.ts#L586) ignored the `options.fetch` usage, so IMO there's no need to write tests around this with regards to RPC consistency.